JS/Lesson35/onder/my-app/src/components/Navbar/index.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useContext, useState } from "react";
+import { NavLink } from "react-router-dom";
+import "./styles.css";
+import { GlobalContext } from "../../Context/GlobalState";
+
+const Navbar = () => {
+  const { setRecipesnew } = useContext(GlobalContext);
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    try {
+      const response = await fetch(
+        `https://www.themealdb.com/api/json/v1/1/search.php?s=${searchQuery}`
+      );
+      const data = await response.json();
+      if (data.meals) {
+        setRecipesnew(data.meals);
```

Comment:
This approach always removes whatever recipe data we had before which may not always be ideal. If you want to preserve previous data, you'd need to create context in a way we did in the lesson (using reducer function would be better in this case). Otherwise, you can use Redux as well but also with reducer function.